ek9000App/src/devEL2XXX.cpp

```diff
@@ -56,38 +57,38 @@ template <class RecordT> static void EL20XX_WriteCallback(CALLBACK* callback) {
 		return;
 	}
 
-	/* Lock & verify mutex */
-	DeviceLock lock(dpvt->pdrv);
+	// Write data to device
+	{
+		DeviceLock lock(dpvt->pdrv);
 
-	if (!lock.valid()) {
-		LOG_ERROR(dpvt->pdrv, "failed to obtain device lock\n");
-		recGblSetSevr(pRecord, WRITE_ALARM, INVALID_ALARM);
-		pRecord->pact = FALSE;
-		return;
-	}
+		if (!lock.valid()) {
+			LOG_ERROR(dpvt->pdrv, "failed to obtain device lock\n");
+			recGblSetSevr(pRecord, WRITE_ALARM, INVALID_ALARM);
+			pRecord->pact = FALSE;
+			return;
+		}
 
-	const bool mbbo = util::is_same<mbboDirectRecord, RecordT>::value;
+		const bool mbbo = util::is_same<mbboDirectRecord, RecordT>::value;
 
-	uint16_t buf[32];
-	const uint16_t length = get_nobt(pRecord); // Will be 1 for bo, nobt for mbboDirect
-	if (mbbo) {
-		// Inflate packed bit data into the input buffer
-		for (int i = 0; i < length; ++i) {
-			buf[i] = (pRecord->rval & (1 << i)) >> i;
+		uint16_t buf[32];
+		const uint16_t length = get_nobt(pRecord); // Will be 1 for bo, nobt for mbboDirect
+		if (mbbo) {
+			// Inflate packed bit data into the input buffer
+			for (int i = 0; i < length; ++i) {
+				buf[i] = (pRecord->rval & (1 << i)) >> i;
+			}
+		}
+		else {
+			buf[0] = pRecord->val;
 		}
-	}
-	else {
-		buf[0] = pRecord->val;
-	}
-
-	const uint16_t addr = mbbo ? dpvt->pterm->m_outputStart - 1 : dpvt->pterm->m_outputStart + (dpvt->channel - 2);
 
-	/* Write to buffer */
-	/** The logic here: channel - 1 for a 0-based index, and subtract another 1 because modbus coils start at 0, and
-	 * inputStart is 1-based **/
-	int status = dpvt->pterm->doEK9000IO(MODBUS_WRITE_MULTIPLE_COILS, addr, buf, length);
+		const uint16_t addr = mbbo ? dpvt->pterm->m_outputStart - 1 : dpvt->pterm->m_outputStart + (dpvt->channel - 2);
 
-	lock.unlock();
+		/* Write to buffer */
+		/** The logic here: channel - 1 for a 0-based index, and subtract another 1 because modbus coils start at 0, and
```

Comment:
clang-format failing here (https://github.com/slaclab/epics-ek9000/actions/runs/5480811470/jobs/9984400579?pr=47)

Tangential but would you consider adding a pre-commit hook for clang-format?

Reply:
I'm actually considering removing it :grin: Tends to be difficult to tune it to format things as you'd like (see the array in #46 for what I mean by that...)

Reply:
Oh, hmm... Yeah, that array is particularly ugly.

Couldn't you just add in `clang-format: off` blocks for those tricky sections?
I think it's more often better to leave formatting to an automated tool, but that's just my opinion

Reply:
Yeah it depends. I like it for repos with many contributors as it helps keep style consistent. If you think it's good to keep, I'll keep it.

Reply:
True about multi-contributor repos. Also good for consistency with diffs, if used before each commit at least.

I think it'd be good to keep it for now and we can revisit it later